### PR TITLE
Add tests & bugfixes for user router

### LINKED ISSUE #94: Tests for user service

### PR DESCRIPTION
**Pull request by Timothy-Gonzalez**
Resolves #94

This covers most of the code and edge cases that I think are reasonable to test. Only things not covered are mongod errors, but those are harder to test.

Tests for:
- [x] GET /user/qr/
- [x] GET /user/qr/:USERID/
- [x] GET /user/
- [x] GET /user/:USERID/
- [x] Remove POST /user/

Also applies a few misc bug fixes, namely typo stuff & returning wrong values.

[Coverage from latest commit](https://github.com/HackIllinois/adonix/actions/runs/6605736630/job/17941216810):
```
 src/services/user       |     100 |       95 |     100 |     100 |                                                                                                                                                                                 
  user-router.ts         |     100 |       95 |     100 |     100 | 111         
```

### CODE REVIEW

File: src/services/user/user-router.ts [hunk 1]
```diff
@@ -128,7 +128,7 @@ userRouter.get("/:USERID", strongJwtVerification, async (req: Request, res: Resp
         if (userInfo) {
             return res.status(Constants.SUCCESS).send(userInfo);
         } else {
-            return res.status(Constants.INTERNAL_ERROR).send({ error: "UserNotFound" });
+            return res.status(Constants.NOT_FOUND).send({ error: "UserNotFound" });
```
Comment (AydanPirani) [inline src/services/user/user-router.ts]: This shouldn't be 404 - 404 is reserved for API endpoint not found. Try 400 instead (invalid request/UserNotFound).

File: src/services/user/user-router.test.ts [hunk 2]
```diff
@@ -21,41 +149,100 @@ describe("GET /", () => {
 
     it("gives an not found error for an non-existent user", async () => {
         await Models.UserInfo.deleteOne({
-            userId: TESTER.id,
+            userId: TESTER_USER.userId,
         });
 
-        const response = await getAsAttendee("/user/").expect(400);
+        const response = await getAsAttendee("/user/").expect(404);
```
Comment (AydanPirani) [inline src/services/user/user-router.test.ts]: 404 -> 400?

File: src/services/user/user-router.test.ts [hunk 3]
```diff
@@ -21,41 +149,100 @@ describe("GET /", () => {
...
+
+        expect(JSON.parse(response.text)).toHaveProperty("error", "Forbidden");
+    });
+
+    it("gives an not found error for a non-existent user", async () => {
+        await Models.UserInfo.deleteOne({
+            userId: OTHER_USER.userId,
+        });
+
+        const response = await getAsStaff(`/user/${OTHER_USER.userId}/`).expect(404);
```
Comment (AydanPirani) [inline src/services/user/user-router.test.ts]: same here

File: src/services/user/user-router.ts [hunk 4]
```diff
@@ -163,7 +163,7 @@ userRouter.get("/", strongJwtVerification, async (_: Request, res: Response) =>
     if (user) {
         return res.status(Constants.SUCCESS).send(user);
     } else {
-        return res.status(Constants.BAD_REQUEST).send({ error: "UserNotFound" });
+        return res.status(Constants.NOT_FOUND).send({ error: "UserNotFound" });
```
Comment (AydanPirani) [inline src/services/user/user-router.ts]: 404->400

File: src/services/user/user-router.test.ts [hunk 5]
```diff
@@ -21,41 +149,100 @@ describe("GET /", () => {
...
+        const response = await getAsStaff(`/user/${OTHER_USER.userId}/`).expect(200);
+
+        expect(JSON.parse(response.text)).toMatchObject(OTHER_USER);
+    });
+});
+
+describe("POST /", () => {
+    it("gives an unauthorized error for an non-staff user", async () => {
+        const response = await postAsAttendee("/user/").send(JSON.stringify(TESTER_USER_WITH_NEW_EMAIL)).expect(403);
+
```
Comment (AydanPirani) [inline src/services/user/user-router.test.ts]: Why does this throw? If we're a user, we should be able to create via auth, no?

File: src/services/user/user-router.test.ts [hunk 6]
```diff
@@ -21,41 +149,100 @@ describe("GET /", () => {
...
+        expect(JSON.parse(response.text)).toMatchObject(NEW_USER);
+
+        const stored = await Models.UserInfo.findOne({
+            userId: NEW_USER.userId,
         });
+
+        expect(stored).toMatchObject(NEW_USER);
+    });
+    it("updates a user for an staff user", async () => {
+        const response = await postAsStaff("/user/").send(JSON.stringify(TESTER_USER_WITH_NEW_EMAIL)).expect(200);
```
Comment (AydanPirani) [inline src/services/user/user-router.test.ts]: Why `post` and not `put` for udpates?

**Review by AydanPirani — CHANGES_REQUESTED** [review]
Had some questions + suggested changes with HTTP codes

File: src/services/user/user-router.ts (hunk 1 above)
Comment (Timothy-Gonzalez) [reply src/services/user/user-router.ts]: Overruled

File: src/services/user/user-router.test.ts (hunk 2 above)
Comment (Timothy-Gonzalez) [reply src/services/user/user-router.test.ts]: Overruled

File: src/services/user/user-router.test.ts (hunk 3 above)
Comment (Timothy-Gonzalez) [reply src/services/user/user-router.test.ts]: Overruled

File: src/services/user/user-router.ts (hunk 4 above)
Comment (Timothy-Gonzalez) [reply src/services/user/user-router.ts]: Overruled

File: src/services/user/user-router.test.ts (hunk 5 above)
Comment (Timothy-Gonzalez) [reply src/services/user/user-router.test.ts]: Removed POST

File: src/services/user/user-router.test.ts (hunk 6 above)
Comment (Timothy-Gonzalez) [reply src/services/user/user-router.test.ts]: Removed POST

**Review by AydanPirani — APPROVED** [review]
lgtm! one thing we should do - add in a status code at some point
